sklearn/metrics/tests/test_common.py

```diff
@@ -583,8 +583,8 @@ def _require_positive_targets(y1, y2):
 def _require_log1p_targets(y1, y2):
     """Make targets strictly larger than -1"""
     offset = abs(min(y1.min(), y2.min())) - 0.99
-    y1 = y1.astype(float)
-    y2 = y2.astype(float)
+    y1 = y1.astype(np.float64)
+    y2 = y2.astype(np.float64)
```

Comment:
Why do we need this change?

Reply:
This was an error I made in a previous PR #29709. Through my involvement in the scikit-learn community, I learned that it’s preferable to use explicit precision here rather than relying on implicit type promotion rules. I took the opportunity in this PR to correct that earlier mistake.